dev/io.openliberty.microprofile.reactive.messaging.internal_fat/fat/src/io/openliberty/microprofile/reactive/messaging/fat/kafka/metrics/MetricsTest.java

```diff
@@ -0,0 +1,151 @@
+                                                                           MetricsDeliveryBean.METRICS_OUTGOING, MetricsTestServlet.OUTGOING_TOPIC))
+                        .include(ConnectorProperties.simpleIncomingChannel(KafkaTests.connectionProperties(), ConnectorProperties.DEFAULT_CONNECTOR_ID,
+                                                                           MetricsReceptionBeanForMessages.CHANNEL_IN, APP_NAME, MetricsTestServlet.OUTGOING_TOPIC));
+
+        WebArchive war = ShrinkWrap.create(WebArchive.class, APP_NAME + ".war")
+                        .addAsLibraries(kafkaClientLibs())
+                        .addAsManifestResource(kafkaPermissions(), "permissions.xml")
+                        .addClasses(MetricsTestServlet.class, MetricsDeliveryBean.class, MetricsReceptionBeanForPayloads.class, MetricsReceptionBeanForMessages.class)
+                        .addPackage(KafkaTestConstants.class.getPackage())
+                        .addPackage(AbstractKafkaTestServlet.class.getPackage())
+                        .addAsResource(appConfig, "META-INF/microprofile-config.properties");
+
+        ShrinkHelper.exportDropinAppToServer(server, war, SERVER_ONLY);
+        server.startServer();
+    }
+
+    @Test
+    public void testDeliverPayload() throws Exception {
+        // Test that metrics are updated when a payload is delivered via an emitter
+        runTest(server, APP_NAME + "/MetricsTest", "emitterDeliverPayload");
+        HashMap<String, Integer> metrics = getMetrics(server);
+
+        assertThat("5 messages not detected on channel", metrics.get("metrics-incoming"), equalTo(5));
+        assertThat("Outgoing and incoming channel counts are not the same",
+                   metrics.get(MetricsTestServlet.EMITTER_OUTGOING_CHANNEL),
+                   equalTo(metrics.get(MetricsReceptionBeanForPayloads.CHANNEL_IN)));
+    }
+
+    @Test
+    public void testDeliverMessage() throws Exception {
+        // Test that metrics are updated when a message is delivered via delivery and reception beans
+        runTest(server, APP_NAME + "/MetricsTest", "deliverMessage");
+        HashMap<String, Integer> metrics = getMetrics(server);
+
+        assertThat("5 messages not detected on channel", metrics.get("metrics-outgoing"), equalTo(5));
+        assertThat("Outgoing and incoming channel counts are not the same",
+                   metrics.get(MetricsDeliveryBean.METRICS_OUTGOING),
+                   equalTo(metrics.get(MetricsReceptionBeanForMessages.CHANNEL_IN)));
+    }
+
+    @AfterClass
+    public static void teardown() throws Exception {
+        try {
+            server.stopServer();
+        } finally {
+            KafkaUtils.deleteKafkaTopics(KafkaTests.getAdminClient());
+        }
+    }
+
+    public static HashMap<String, Integer> getMetrics(LibertyServer server) throws Exception {
```

Comment:
If this is to be common across tests, it should be in its own common class, not in one test class and then reused by another. while we only have two tests at the moment here, if we ever expand this then it is likely to start to cause issues. So it being static in the common class taking the server as its argument is correct, this could lead to some unintended issues, even replacing server with URL